ORC: Switch tests to Junit5

skytin1004 opened the pull request:
This pull request is related to the issue "Move JUnit4 tests to JUnit5 https://github.com/apache/iceberg/issues/7160".
I've switched the files in ORC pakage to JUnit5.

I've checked build successfully.

nastra commented on build.gradle:
```diff
@@ -957,4 +960,3 @@ apply from: 'jmh.gradle'
 apply from: 'baseline.gradle'
 apply from: 'deploy.gradle'
 apply from: 'tasks.gradle'
-
```
nit: please revert this change

skytin1004 replied:
I've reverted this change

nastra commented on orc/src/test/java/org/apache/iceberg/orc/TestBloomFilter.java:
```diff
@@ -54,12 +52,11 @@ public class TestBloomFilter {
           required(102, "price", Types.DoubleType.get()));
 
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
+  @TempDir File testFile;
```
nit: can be made private

nastra commented on orc/src/test/java/org/apache/iceberg/orc/TestOrcDataReader.java:
```diff
@@ -43,13 +43,13 @@
 public class TestOrcDataReader implements WithAssertions {
-  @ClassRule public static TemporaryFolder temp = new TemporaryFolder();
+
+  @TempDir static File temp;
```
nit: can probably be made private

skytin1004 replied:
I've changed 'temp' to private

nastra commented on orc/src/test/java/org/apache/iceberg/orc/TestOrcDataWriter.java:
```diff
@@ -62,9 +61,9 @@ public class TestOrcDataWriter {
   private List<Record> records;
 
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
+  @TempDir File temp;
```
private

skytin1004 replied:
I've changed 'temp' to private

nastra commented on orc/src/test/java/org/apache/iceberg/orc/TestOrcDataWriter.java:
```diff
@@ -141,7 +137,7 @@ public void testUsingFileIO() throws IOException {
     // use a scheme `dummy` that is not handled. Note that Hadoop 2.7.3 throws IOException
     // while latest Hadoop versions throw UnsupportedFileSystemException (extends IOException)
-    ProxyOutputFile outFile = new ProxyOutputFile(Files.localOutput(temp.newFile()));
+    ProxyOutputFile outFile = new ProxyOutputFile(Files.localOutput(String.valueOf(temp.isFile())));
```
this seems wrong. This expects a file and not a "true"/"false" string

skytin1004 replied:
I changed the variable to go inside to fit the context. `String.valueOf(temp.isFile())` to `temp`

nastra commented on orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java:
```diff
@@ -55,9 +54,9 @@ public class TestOrcDeleteWriters {
   private List<Record> records;
 
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
+  @TempDir File temp;
```
private

skytin1004 replied:
I've changed 'temp' to private

nastra commented on orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java:
```diff
@@ -73,7 +72,7 @@ public void createDeleteRecords() {
   @Test
   public void testEqualityDeleteWriter() throws IOException {
-    File deleteFile = temp.newFile();
+    File deleteFile = temp;
```
we can probably remove `deleteFile` alltogether

skytin1004 replied:
I've removed the deleteFile

nastra commented on orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java:
```diff
@@ -106,12 +104,14 @@ public void testEqualityDeleteWriter() throws IOException {
   @Test
   public void testPositionDeleteWriter() throws IOException {
-    File deleteFile = temp.newFile();
+    File deleteFile = temp;
```
`deleteFile` can be removed

skytin1004 replied:
I've removed the `deleteFile`

nastra commented on orc/src/test/java/org/apache/iceberg/orc/TestOrcDeleteWriters.java:
```diff
@@ -163,13 +162,14 @@ public void testPositionDeleteWriter() throws IOException {
   @Test
   public void testPositionDeleteWriterWithEmptyRow() throws IOException {
-    File deleteFile = temp.newFile();
+    File deleteFile = temp;
```
same here

skytin1004 replied:
I've removed the `deleteFile`

nastra commented on orc/src/test/java/org/apache/iceberg/orc/TestTableProperties.java:
```diff
@@ -50,7 +49,8 @@ public class TestTableProperties {
 
-  @ClassRule public static final TemporaryFolder TEMPORARY_FOLDER = new TemporaryFolder();
+  @TempDir File folder;
+  @TempDir File testFile;
```
both can be private

skytin1004 replied:
I've changed 'folder','testFile' to private

nastra commented:
@skytin1004 there's a binary file in the PR named [orc/false](https://github.com/apache/iceberg/pull/7954/files#diff-65316bcea8c0ad32a87ef1205a356d75f1eae7fba6be70bdb01b7a286dc5c5bb) that needs to be removed

skytin1004 commented:
Thanks @nastra, I have fixed the code based on your review.

skytin1004 commented:
Thanks @nastra for the detailed review, I have deleted the file [orc/false](https://github.com/apache/iceberg/pull/7954/files#diff-65316bcea8c0ad32a87ef1205a356d75f1eae7fba6be70bdb01b7a286dc5c5bb) and I have modified the code based on your review.
